metaflow/plugins/kfp/argo_utils.py

```diff
@@ -0,0 +1,165 @@
+import math
+import time
+import datetime
+from typing import Optional, Union, Dict, Any
+
+from metaflow.metaflow_config import ARGO_RUN_URL_PREFIX, METAFLOW_RUN_URL_PREFIX
+from metaflow.plugins.kfp.argo_client import ArgoClient
+from metaflow.plugins.kfp.kfp_decorator import KfpException
+from metaflow.plugins.kfp.kfp_utils import _get_kfp_logger
+
+
+logger = _get_kfp_logger()
+
+
+def run_argo_workflow(
+    kubernetes_namespace: str,
+    template_name: str,
+    parameters: Optional[dict] = None,
+    wait_timeout: Union[int, float, datetime.timedelta] = 0,
+    **kwarg,  # Other parameters for wait function
+) -> str:
+    try:
+        # TODO(talebz): add tag of origin-run-id to correlate parent flow
```

Comment:
Do you intend to resolve this before PR merging, or do you have a ticket created?

Reply:
created https://zbrt.atl.zillow.net/browse/AIP-7382